org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/experimental/XMLFormatterJoinCommentLinesTest.java

```diff
@@ -0,0 +1,136 @@
+		String content = "<a>" + lineSeparator() + //
+				"  <!--" + lineSeparator() + //
+				"   line 1" + lineSeparator() + //
+				"   " + lineSeparator() + //
+				"   " + lineSeparator() + //
+				"     line 2" + lineSeparator() + //
+				"   -->" + lineSeparator() + //
+				"</a>";
+		String expected = "<a>" + lineSeparator() + //
+				"  <!-- line 1 line 2 -->" + lineSeparator() + //
+				"</a>";
+
+		SharedSettings settings = new SharedSettings();
+		settings.getFormattingSettings().setJoinCommentLines(true);
+		assertFormat(content, expected, settings);
+	}
+
+	@Test
+	public void testCommentFormatSameLine() throws BadLocationException {
+		String content = "<a>" + lineSeparator() + //
+				" Content" + lineSeparator() + //
+				"</a> <!-- My   Comment   -->";
+		String expected = "<a> Content </a> <!-- My Comment -->";
+
+		SharedSettings settings = new SharedSettings();
+		settings.getFormattingSettings().setJoinCommentLines(true);
+		assertFormat(content, expected, settings);
+	}
+
+	@Test
+	public void testJoinCommentLinesLongWrap() throws BadLocationException {
+		String content = "<a>" + lineSeparator() + //
+				"  Content <!-- comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment comment -->"
+				+ //
+				"</a>";
+		String expected = "<a> Content <!-- comment comment comment comment comment comment comment comment comment"
+				+ lineSeparator() + //
+				"  comment comment comment comment comment comment comment comment comment"
+				+ lineSeparator() + //
+				"  comment comment comment comment comment comment comment -->" + //
+				"</a>";
+		SharedSettings settings = new SharedSettings();
+		settings.getFormattingSettings().setJoinCommentLines(true);
+		assertFormat(content, expected, settings);
+	}
+
+	@Test
+	public void testJoinCommentLinesLongWrapSingleWord() throws BadLocationException {
+		String content = "<a>" + lineSeparator() + //
+				"<!-- commentcommentcommentcomment commentcommentcommentcomment commentcommentcommentcomment commentcommmentcommentcommentcomment commentcommentcommentscommentcomment commentcommentcommentscommentcomment commentcommentcomments -->"
+				+ //
+				"</a>";
+		String expected = "<a>" + lineSeparator() + //
+				"  <!-- commentcommentcommentcomment commentcommentcommentcomment" + lineSeparator() + //
+				"  commentcommentcommentcomment commentcommmentcommentcommentcomment" + lineSeparator() + //
+				"  commentcommentcommentscommentcomment commentcommentcommentscommentcomment" + lineSeparator() + //
+				"  commentcommentcomments --></a>";
+		SharedSettings settings = new SharedSettings();
+		settings.getFormattingSettings().setJoinCommentLines(true);
+		assertFormat(content, expected, settings);
+	}
+
+	private static void assertFormat(String unformatted, String expected, SharedSettings sharedSettings)
```

Comment:
Remove this method, with experimental formatter, we need to test the text edit (if text edit is empty or not).